controllers/util.go

```diff
@@ -49,6 +50,21 @@ func (c *ApiController) ResponseError(error string, data ...interface{}) {
 	c.ResponseJsonData(resp, data...)
 }
 
+// ResponseErrorWithStatus ...
+func (c *ApiController) ResponseErrorWithStatus(status int, error string) {
+	resp := &Error{Code: status, Message: error}
+
+	http.Error(c.Ctx.ResponseWriter, error, status)
+
+	c.Data["json"] = resp
+	c.ServeJSON()
```

Comment:
is it works right? looks like http.Error and c.ServerJSON do same work, mb u.Ctx.Output.Status = http.StatusNotFound instead http.Error will better here?

Reply:
yep, it's a better way thx